src/librustc_mir/build/mod.rs

```diff
@@ -675,8 +675,12 @@ impl<'a, 'gcx, 'tcx> Builder<'a, 'gcx, 'tcx> {
             // If this is a simple binding pattern, give the local a nice name for debuginfo.
             let mut name = None;
             if let Some(pat) = pattern {
-                if let hir::PatKind::Binding(_, _, ident, _) = pat.node {
-                    name = Some(ident.name);
+                if let hir::PatKind::Binding(binding_annotation, _, ident, _) = pat.node {
+                    if binding_annotation == hir::BindingAnnotation::Unannotated
+                        || binding_annotation == hir::BindingAnnotation::Mutable
```

Comment:
I don't quite understand this part — does this not affect debuginfo in negative ways or something?

Also, a nit: maybe we should use a match here?

Reply:
I presume this corresponds to 

> Change mir generation so that the parameter variable doesn't get a name when a ref pattern is used as an argument

? 

In that case, I would want to see some comments to that effect...